src/saml.py

```diff
@@ -66,6 +84,30 @@ def metadata_url(self) -> str:
         """
         return self._charm_state.metadata_url
 
+    @cached_property
+    def _is_certificate_valid(self):
+        """Validate the public certificate for the metadata URL.
+
+        Returns:
+            True if the certificate matches the one provided.
+
+        Raises:
+            TimeoutError: if the connection can't be established.
+        """
```

Comment:
I though the certificate validation was validating the certificate provided inside the metadata, where's validating here the certificate used by the server to serve the metadata.
This shouldn't be needed as this certificate must be signed by a CA trusted by the OS